### PR TITLE
rgw: fix broken /crossdomain.xml, /info and /healthcheck of Swift API.

### PR DESCRIPTION
**Pull request by rzarzynski**
Fixes: http://tracker.ceph.com/issues/19520
Signed-off-by: Radoslaw Zarzynski <rzarzynski@mirantis.com>

### CODE REVIEW

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -374,8 +374,13 @@ int rgw_build_bucket_policies(RGWRados* store, struct req_state* s)
   if(s->dialect.compare("s3") == 0) {
     s->bucket_acl = new RGWAccessControlPolicy_S3(s->cct);
   } else if(s->dialect.compare("swift")  == 0) {
-    s->user_acl = std::unique_ptr<RGWAccessControlPolicy>(
-        new RGWAccessControlPolicy_SWIFTAcct(s->cct));
+    /* We aren'tallocating the account policy for those operations using
```
Comment (yehudasa) [inline src/rgw/rgw_op.cc]: missing a space here at the comment

Comment (rzarzynski) [inline src/rgw/rgw_op.cc]: Fixed.

**Comment by yehudasa** [comment]
@rzarzynski how did you test it?

**Comment by rzarzynski** [comment]
@yehudasa: I've locally verified the branch using Tempest and s3-tests. Neither of them has found any regression.

Additionally, Tempest confirmed the *slash features* became operational:
```
$ ./run_tempest.sh -V "tempest.api.object_storage.test_crossdomain.CrossdomainTest.test_get_crossdomain_policy|tempest.api.object_storage.test_healthcheck.HealthcheckTest.test_get_healthcheck|tempest.api.object_storage.test_account_services.AccountTest.test_list_extensions"
WARNING: This script is deprecated and will be removed in the near future. Please migrate to tempest run or another method of launching a test runner
running=OS_STDOUT_CAPTURE=${OS_STDOUT_CAPTURE:-1} \
OS_STDERR_CAPTURE=${OS_STDERR_CAPTURE:-1} \
OS_TEST_TIMEOUT=${OS_TEST_TIMEOUT:-500} \
OS_TEST_LOCK_PATH=${OS_TEST_LOCK_PATH:-${TMPDIR:-'/tmp'}} \
${PYTHON:-python} -m subunit.run discover -t ${OS_TOP_LEVEL:-./} ${OS_TEST_PATH:-./tempest/test_discover} --list 
running=OS_STDOUT_CAPTURE=${OS_STDOUT_CAPTURE:-1} \
OS_STDERR_CAPTURE=${OS_STDERR_CAPTURE:-1} \
OS_TEST_TIMEOUT=${OS_TEST_TIMEOUT:-500} \
OS_TEST_LOCK_PATH=${OS_TEST_LOCK_PATH:-${TMPDIR:-'/tmp'}} \
${PYTHON:-python} -m subunit.run discover -t ${OS_TOP_LEVEL:-./} ${OS_TEST_PATH:-./tempest/test_discover}  --load-list /tmp/tmpfFAnTb
running=OS_STDOUT_CAPTURE=${OS_STDOUT_CAPTURE:-1} \
OS_STDERR_CAPTURE=${OS_STDERR_CAPTURE:-1} \
OS_TEST_TIMEOUT=${OS_TEST_TIMEOUT:-500} \
OS_TEST_LOCK_PATH=${OS_TEST_LOCK_PATH:-${TMPDIR:-'/tmp'}} \
${PYTHON:-python} -m subunit.run discover -t ${OS_TOP_LEVEL:-./} ${OS_TEST_PATH:-./tempest/test_discover}  --load-list /tmp/tmp_3lTEt
running=OS_STDOUT_CAPTURE=${OS_STDOUT_CAPTURE:-1} \
OS_STDERR_CAPTURE=${OS_STDERR_CAPTURE:-1} \
OS_TEST_TIMEOUT=${OS_TEST_TIMEOUT:-500} \
OS_TEST_LOCK_PATH=${OS_TEST_LOCK_PATH:-${TMPDIR:-'/tmp'}} \
${PYTHON:-python} -m subunit.run discover -t ${OS_TOP_LEVEL:-./} ${OS_TEST_PATH:-./tempest/test_discover}  --load-list /tmp/tmp8zwzup
{0} tempest.api.object_storage.test_healthcheck.HealthcheckTest.test_get_healthcheck [0.004530s] ... ok
{1} tempest.api.object_storage.test_crossdomain.CrossdomainTest.test_get_crossdomain_policy [0.004683s] ... ok
{2} tempest.api.object_storage.test_account_services.AccountTest.test_list_extensions [0.002800s] ... ok

======
Totals
======
Ran: 3 tests in 4.0000 sec.
 - Passed: 3
 - Skipped: 0
 - Expected Fail: 0
 - Unexpected Success: 0
 - Failed: 0
Sum of execute time for each test: 0.0120 sec.

==============
Worker Balance
==============
 - Worker 0 (1 tests) => 0:00:00.004530
 - Worker 1 (1 tests) => 0:00:00.004683
 - Worker 2 (1 tests) => 0:00:00.002800
```

**Comment by rzarzynski** [comment]
@yehudasa: this branch has been tested in following Teuthology runs:
* http://pulpito.ceph.com/rzarzynski-2017-04-29_14:39:05-rgw-wip-rzarzynski-testing---basic-mira/,
* http://pulpito.ceph.com/rzarzynski-2017-04-30_12:33:39-rgw-wip-rzarzynski-testing---basic-mira/.

The results look good. The failures have been caused by the well-known `strange leak of std::string memory from md_config_t seen in radosgw` signalised by Valgrind.

**Pull request merged**